controllers/kserve_inferenceservice_controller.go

```diff
@@ -644,5 +665,13 @@ func (r *OpenshiftInferenceServiceReconciler) ReconcileKserveInference(ctx conte
 	if err != nil {
 		return err
 	}
+
+	log.Info("Reconciling Generic Route for Kserve InferenceService")
+	kisvcRouteReconciler := reconcilers.NewKserveInferenceServiceRouteReconciler(r.Client, r.Scheme, ctx, log, inferenceService)
+	err = kisvcRouteReconciler.Reconcile()
```

Comment:
I haven't tested, but there seems to be a potential race condition here.

If `kserve-controller` reconciliation is delayed (i.e., heavy load), it is possible that `odh-model-controller` tries to reconcilie before `kserve-controller`. URLs will still not be set and `odh-model-controller` should re-try later.

I cannot spot any part where a retry is scheduled. So, I think will work only if order of events is _ideal_.

Please, cover the case where KServe reconciliation is delayed (for any reason). Or, please, correct me if I'm wrong and I was unable to spot the schedule of the retry.

Reply:
@israel-hdez nice catch.
Currently we don't have any provision in our code to send back schedule interval from function call. If I go to implement it right now then it would cause lot of changes and maybe hard to define boundaries of this partical task.

For now, I just retrun the `error` if ServiceURL is not available which would keep retrigggering the reconciler untill ServiceURL is available. 
I would raise a new ticket to support schduler functionality in our code.

Reply:
I had created below ticket to add scheduler interval support in odh-model-controller.
https://github.com/opendatahub-io/odh-model-controller/issues/89